### PR TITLE
Improve running time of TestSchedulerWithVolumeBinding

### PR DESCRIPTION
**Pull request by alculquicondor**
Only wait for finished binding or error, but not both

**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

The scheduler would produce either, but not both. No need for busy waiting.

The running time of the test goes from 17s down to 1s.

**Which issue(s) this PR fixes**:

Somewhat related to investigations in #93890

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
cc @cofyc 

/assign @Huang-Wei 

Thanks @liggitt for the suggestion

File: pkg/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -953,27 +953,19 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
...
+			// Wait for scheduling to return an error or succeed binding.
 			select {
 			case err := <-errChan:
 				if item.expectError == nil || !reflect.DeepEqual(item.expectError.Error(), err.Error()) {
 					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectError, err)
 				}
-			case <-time.After(chanTimeout):
-				if item.expectError != nil {
-					t.Errorf("did not receive error after %v", chanTimeout)
-				}
```
Comment (liggitt) [inline pkg/scheduler/scheduler_test.go:965]: move this to the `<-time.After(chanTimeout)` case

File: pkg/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -953,27 +953,19 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
...
-			// Wait for pod to succeed binding
-			select {
 			case b := <-bindingChan:
 				if !reflect.DeepEqual(item.expectPodBind, b) {
 					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectPodBind, b)
 				}
 			case <-time.After(chanTimeout):
 				if item.expectPodBind != nil {
-					t.Errorf("did not receive pod binding after %v", chanTimeout)
+					t.Errorf("did not receive pod binding or error after %v", chanTimeout)
```
Comment (Huang-Wei) [inline pkg/scheduler/scheduler_test.go:968]: We should also error out below `if item.expectError != nil`.

File: pkg/scheduler/scheduler_test.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/scheduler_test.go:965]: Better done after the select.

File: pkg/scheduler/scheduler_test.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/scheduler/scheduler_test.go:968]: we don't actually need to check anything in the case of timeout. It's an error regardless.

**Comment by alculquicondor** [comment]
@liggit do you want to include it in the milestone?

**Comment by liggitt** [comment]
at this point, I'm only including flake fixes in the milestone, and am leaving test runtime improvements for when master reopens

File: pkg/scheduler/scheduler_test.go [hunk 3]
```diff
@@ -953,28 +953,26 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
...
+				gotBind *v1.Binding
+			)
 			select {
-			case err := <-errChan:
-				if item.expectError == nil || !reflect.DeepEqual(item.expectError.Error(), err.Error()) {
-					t.Errorf("err \nWANT=%+v,\nGOT=%+v", item.expectError, err)
-				}
+			case gotErr = <-errChan:
+			case gotBind = <-bindingChan:
 			case <-time.After(chanTimeout):
```
[the remark this answers is not in the document]
Comment (krmayankk) [reply pkg/scheduler/scheduler_test.go:964]: sounds good, probably its just me then.

**Pull request merged**